network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/PhaseTapChangerImpl.java

```diff
@@ -148,4 +146,35 @@ public static void validateStep(TapChangerStepAttributes step, TapChangerParent
             throw new ValidationException(parent, "step alpha is not set");
         }
     }
+
+    // equals and hashcode are override because of ValidationUtil.checkPhaseTapChangerRegulation that
+    // remove the tap changer of the list of allRegulatingTapChanger
+    // and it does not work if equals and hascode check this.attributesGetter
+    @Override
+    public boolean equals(Object o) {
```

Comment:
why checking only these attributes and not the ones of TapChangerAttributes?

Reply:
I think it is enough to identified a tapChanger